webdriver/tests/bidi/browsing_context/set_viewport/set_viewport.py

```diff
@@ -0,0 +1,88 @@
+import pytest
+
+from ... import get_viewport_dimensions
+
+
+@pytest.mark.asyncio
+async def test_set_viewport(bidi_session, new_tab):
+    test_viewport = {"width": 250, "height": 300}
+
+    await bidi_session.browsing_context.set_viewport(
+        context=new_tab["context"],
+        viewport=test_viewport)
+
+    assert await get_viewport_dimensions(bidi_session, new_tab) == test_viewport
+
+
+@pytest.mark.asyncio
+async def test_set_viewport_reset(bidi_session, new_tab):
+    original_viewport = await get_viewport_dimensions(bidi_session, new_tab)
+
+    test_viewport = {"width": 666, "height": 333}
+    await bidi_session.browsing_context.set_viewport(
+      context=new_tab["context"],
+      viewport=test_viewport)
+
+    assert await get_viewport_dimensions(bidi_session, new_tab) == test_viewport
+
+    await bidi_session.browsing_context.set_viewport(
+      context=new_tab["context"],
+      viewport=None)
+    assert await get_viewport_dimensions(bidi_session, new_tab) == original_viewport
+
+
+@pytest.mark.asyncio
+async def test_set_viewport_affects_specific_context(bidi_session, top_context, new_tab):
+    original_viewport = await get_viewport_dimensions(bidi_session, top_context)
+
+    test_viewport = {"width": 333, "height": 333}
```

Comment:
You updated the other case but here the height and width are still the same.

Reply:
My comment is not about the new tab. The point is if due to some reason the remote end takes eg the value for width and applies it to both width and height we won't notice that if the value is equal.

Reply:
This is covered by the other test (test_set_viewport), right?

Reply:
But can you guarantee that there is no special situation when this could happen? Most likely not. It doesn't hurt to say eg. `499, 599` as for other tests.